tests/PHPStan/Rules/Arrays/data/array-unpacking.php

```diff
@@ -18,12 +18,18 @@ function stringKeyedArray(array $bar)
 	$baz = [...$bar];
 }
 
-/** @param array<array-key, string> $bar */
+/** @param array<int|string, string> $bar */
 function unionKeyedArray(array $bar)
 {
 	$baz = [...$bar];
 }
 
+/** @param array<array-key, string> $bar */
+function benevolentUnionKeyedArray(array $bar)
+{
+	$baz = [...$bar];
```

Comment:
Add new tests at the end of the file, otherwise it's hard to review the changes.

Reply:
The diff should be better now